packages/ui/src/AddressView/index.tsx

```diff
@@ -4,49 +4,84 @@ import { bech32mAddress } from '@penumbra-zone/bech32m/penumbra';
 import { CopyToClipboardButton } from '../CopyToClipboardButton';
 import { AddressIcon } from './AddressIcon';
 import { Text } from '../Text';
+import { Density, useDensity } from '../utils/density';
 
 export interface AddressViewProps {
   addressView: AddressView | undefined;
   copyable?: boolean;
   hideIcon?: boolean;
+  truncate?: boolean;
 }
 
+export const getIconSize = (density: Density): number => {
+  if (density === 'compact') {
+    return 16;
+  }
+  if (density === 'slim') {
+    return 12;
+  }
+  return 24;
+};
+
 // Renders an address or an address view.
 // If the view is given and is "visible", the account information will be displayed instead.
 export const AddressViewComponent = ({
   addressView,
   copyable = true,
   hideIcon,
+  truncate = false,
 }: AddressViewProps) => {
+  const density = useDensity();
+
   if (!addressView?.addressView.value?.address) {
     return null;
   }
 
   const addressIndex = getAddressIndex.optional(addressView);
 
-  // a randomized index has nonzero randomizer bytes
+  // A randomized index has nonzero randomizer bytes
   const isRandomized = addressIndex?.randomizer.some(v => v);
 
   const encodedAddress = bech32mAddress(addressView.addressView.value.address);
 
+  // Sub-account selector logic
+  const getAccountLabel = (index: number) =>
+    index === 0 ? 'Main Account' : `Sub-Account ${index}`;
+
   return (
-    <div className='flex items-center gap-2 text-text-primary'>
+    <div className={'flex items-center gap-2 text-text-primary'}>
       {!hideIcon && (
         <div className='shrink'>
-          <AddressIcon address={addressView.addressView.value.address} size={24} />
+          <AddressIcon
+            address={addressView.addressView.value.address}
+            size={getIconSize(density)}
+          />
         </div>
       )}
 
-      {addressIndex ? (
-        <Text strong truncate>
-          {isRandomized && 'IBC Deposit Address for '}
-          {`Sub-Account #${addressIndex.account}`}
-        </Text>
-      ) : (
-        <Text technical truncate>
-          {encodedAddress}
-        </Text>
-      )}
+      <div className={truncate ? 'truncate max-w-[150px]' : ''}>
+        {addressIndex ? (
+          density === 'sparse' ? (
+            <Text strong-bold truncate={truncate}>
+              {isRandomized && 'IBC Deposit Address for '}
+              {getAccountLabel(addressIndex.account)}
+            </Text>
+          ) : (
+            <Text small truncate={truncate}>
+              {isRandomized && 'IBC Deposit Address for '}
+              {getAccountLabel(addressIndex.account)}
+            </Text>
```

Comment:
thought: this makes me think of creating an alternative prop that would simply accept a string